### ISSUE TITLE
assets.deploy() is creating one file with 0 bytes

### ISSUE THREAD

**Issue by hgilani** [techpines/asset-rack#28]
I am using following function to deploy assets. But, I always see one file with size of 0 bytes in the container. File gets the correct hashed url though. Tried to understand source but I am not familiar with coffee syntax.

Is this an issue in asset-rack or an issue in my function. Following is my function

```
function memento_io() {
    var assets = new rack.Rack([
        new rack.StaticAssets({
            urlPrefix: '/',
            dirname: __dirname + '/public'
        })
    ]);

    assets.on('complete', function(){
        console.log(assets);
        this.deploy({
            provider: 'rackspace',
            container: 'memento_io',
            username: config.cdn["username"],
            apiKey: config.cdn.api_key,
        }, function(error) {
            if(error) {
                console.log(error);
            }
            console.log(':)');
        });
    });
}
memento_io();
```

**Comment by techpines** [comment]
In 2.x, you don't need to wait for the `complete` event, you can just deploy immediately:

```
assets.deploy
```

So are you saying that all of the other files are uploading correctly, but one out of a group of static files are not uploading?  Can you give a little more information?

**Comment by hgilani** [comment]
Thanks for looking into this. I was deploying the public folder which has following structure

I first tried to deploy public folder with following contents. After deploy I noticed in rackspace web dashboard that "pre_launch.css" has size of 0 bytes.

```
public:
pre_launch.html

public/css:
pre_launch.css
```

Then, I modified another css file in the css folder "pre_launch1.css" and deployed again. This time 2 files had the data and one file was still 0 bytes. 

See attached image of what I see in web dashboard

![zero bytes](https://f.cloud.github.com/assets/824674/157833/b0070ae0-76d0-11e2-8cd3-68b477f24877.png)

**Comment by techpines** [comment]
I was able to replicate this. It always seems to happen with the first asset uploaded.   I tested it out using Amazon S3, and that works totally fine.  The library that I am relying on for uploads is called pkgcloud by nodejitsu.  It's possible they have a bug.  I fixed an Amazon bug of theirs the other week.

I've got a hack that will fix it (upload the first asset twice).  

I just published the fix, so give it a try:

```
asset-rack@2.0.7
```
